controllers/user.js

```diff
@@ -1,15 +1,132 @@
 const express = require('express');
 const { encryptPassword, comparePassword } = require('../infra/encryption');
-
+const { auth } = require("../infra/jwt/index")
+const User = require("../models/user")
 const router = express.Router();
-
+const bcrypt = require("bcryptjs")
 // --- Routes
 
-router.get('/', (req, res) => res.send('Hello World!'));
-// post
-// put
-// delete
+// Endpoint:  /api/users
+//Creating a user
+router.post("/",async (req,res)=>{
+
+    const user = new User(req.body);
```

Comment:
Just make sure you apply a check to find whether an existing user with the same credentials exists in the db or not before making the new user.